tests/flytekit/common/workflows/sagemaker.py

```diff
@@ -0,0 +1,203 @@
+        extra_input1,
+        extra_input2,
+        model,
+        extra_output1,
+    ):
+
+    with train as reader:
+        train_df = reader.read(concat=True)
+        dtrain_x = xgb.DMatrix(train_df[:-1])
+        dtrain_y = xgb.DMatrix(train_df[-1])
+    with validation as reader:
+        validation_df = reader.read(concat=True)
+        dvalidation_x = xgb.DMatrix(validation_df[:-1])
+        dvalidation_y = xgb.DMatrix(validation_df[-1])
+
+    static_hyperparameters["num_round"] = extra_input1 + extra_input2
+
+    my_model = xgb.XGBModel(**static_hyperparameters)
+
+    my_model.fit(dtrain_x,
+                 dtrain_y,
+                 eval_set=[(dvalidation_x, dvalidation_y)],
+                 eval_metric=sample_eval_function)
+
+    model.set(my_model)
+    extra_output1.set(my_model.evals_result())
+
+
+@workflow_class
+class SageMakerSimpleWorkflow(object):
+    a = Input(Types.Integer, required=True, help="Test integer with no default")
+    b = Input(Types.Integer, required=False, default="1", help="Test integer with default")
+
+    my_simple_xgboost_trainingjob_task = simple_xgboost_trainingjob_task(
+        train='s3://my-bucket/training.csv',
+        validation='s3://my-bucket/validation.csv',
+        static_hyperparameters=example_hyperparams,
+    )
+
+    my_simple_hpojob_task = simple_hpojob_task(
+        trainingjob_task=my_simple_xgboost_trainingjob_task,  # TODO: this doesn't seem totally reasonable
+        hyperparameter_ranges={   # hyperparameters included here will automatically override the static counterpart
+            # https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost_hyperparameters.html
+            "num_rounds": IntegerParameterRange(min_value="1", max_value="100", scaling_type="LOGARITHMIC"),
+            "max_leaves": IntegerParameterRange(min_value="0", max_value="5", scaling_type="LINEAR"),
+            "sketch_eps": ContinuousParameterRange(min_value="0.01", max_value="0.05", scaling_type="AUTO"),
+            "tree_method": CategoricalParameterRange(values=["hist", "exact"]) # TODO: sagemaker doesn't accept gpu_hist. Figure out how to do gpu_hist
+        },
+        hyperparameter_tuning_strategy=HPOJobTuningStrategy.BAYESIAN,
+        hyperparameter_tuning_objective=HPOJobObjective(type="MINIMIZE", metric_name="validation:error"),
+        trainingjob_early_stopping_type="AUTO",
+    )
+
+    my_custom_trainingjob_task = custom_trainingjob_task(
+        extra_input1=a,
+        extra_input2=b,
+        train='s3://my-bucket/training.csv',
+        validation='s3://my-bucket/validation.csv',
+        static_hyperparameters=example_hyperparams,
+    )
+
+    my_custom_task = simple_hpojob_task(
+        trainingjob_task=my_custom_trainingjob_task,  # TODO: this doesn't seem totally reasonable
+        hyperparameter_ranges={  # hyperparameters included here will automatically override the static counterpart
+            # TODO: Need to figure out how to do decide precedence
```

Comment:
what does the TODO mean?

Reply:
This is like a reminder for myself that the implementation of this might be tricky for BYOC use case. 

For example, if in `custom_trainingjob_task` user has some code to calculate the value of a hyperparameter, then what is specified might be overriden if not implemented carefully